storagemarket/impl/provider_states.go

```diff
@@ -221,15 +222,21 @@ func (p *Provider) sealing(ctx context.Context, deal MinerDeal) (func(*MinerDeal
 }
 
 func (p *Provider) complete(ctx context.Context, deal MinerDeal) (func(*MinerDeal), error) {
-	// TODO: observe sector lifecycle, status, expiration..
-        err := p.fs.Delete(deal.PiecePath)
-        if err != nil {
-                return nil, err
-        }
+	err := p.fs.Delete(deal.PiecePath)
+	if err != nil {
+		return nil, err
+	}
 	sectorID, offset, length, err := p.spn.LocatePieceForDealWithinSector(ctx, deal.DealID)
 	if err != nil {
 		return nil, err
 	}
+	// TODO: Record actual block locations for all CIDs in piece by improving car writing
```

Comment:
SHould this be captured in an issue?

Reply:
yes, it will be soon, that's the intent.